src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -327,22 +334,56 @@ private void fileAclOp(AclOp op, Dataset dataset, List<String> fileIds, Map<IamR
     }
 
     /**
-     * Extract the bucket from a gs path.
+     * Extract the path portion (everything after the bucket name and it's trailing slash) of a gs path.
      */
-    public static String extractBucketFromPath(final String path) {
-        if (!path.startsWith("gs://") && StringUtils.countMatches(path, ',') < 3) {
-            throw new PdaoException(String.format("Bad GS path %s", path));
-        }
-        // Given a string that looks like "gs://some bucket/pat1/path2" (format verified above), this will extract
-        // "some bucket" and return
-        return path.substring(5).split("/")[0];
+    private static String extractFilePathInBucket(final String path, final String bucketName) {
+        return StringUtils.removeStart(path, String.format("gs://%s/", bucketName));
     }
 
     /**
-     * Extract the path portion (everything after the bucket name and it's trailing slash) of a gs path.
+     * Represents a way to access objects in GCS buckets.
      */
-    public static String extractFilePathInBucket(final String path,
-                                                 final String bucketName) {
-        return StringUtils.removeStart(path, String.format("gs://%s/", bucketName));
+    public static class GcsLocator {
+        private final String bucket;
+        private final String path;
+
+        public GcsLocator(final String bucket, final String path) {
+            this.bucket = bucket;
+            this.path = path;
+        }
+
+        public String getBucket() {
+            return bucket;
+        }
+
+        public String getPath() {
+            return path;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            GcsLocator locator = (GcsLocator) o;
+
+            return new EqualsBuilder()
```

Comment:
Is this better than `Objects.equals(bucket, locator.bucket) && Objects.equals(path, locator.path)`?